crypto/src/hash/pedersen/mod.rs

```diff
@@ -0,0 +1,102 @@
+use lambdaworks_math::{
+    cyclic_group::IsGroup,
+    elliptic_curve::short_weierstrass::{
+        curves::stark_curve::StarkCurve, point::ShortWeierstrassProjectivePoint,
+        traits::IsShortWeierstrass,
+    },
+    field::{
+        element::FieldElement, fields::fft_friendly::stark_252_prime_field::Stark252PrimeField,
+    },
+};
+
+pub mod parameters;
+use self::parameters::PedersenParameters;
+
+pub struct Pedersen<EC: IsShortWeierstrass> {
+    params: PedersenParameters<EC>,
+}
+
+impl Default for Pedersen<StarkCurve> {
+    fn default() -> Self {
+        let pedersen_stark_default_params = PedersenParameters::<StarkCurve>::default();
+        Self::new_with_params(pedersen_stark_default_params)
+    }
+}
+
+impl Pedersen<StarkCurve> {
+    pub fn new_with_params(params: PedersenParameters<StarkCurve>) -> Self {
+        Self { params }
+    }
+
+    /// Taken from Jonathan Lei's starknet-rs
+    /// https://github.com/xJonathanLEI/starknet-rs/blob/4ab2f36872435ce57b1d8f55856702a6a30f270a/starknet-crypto/src/pedersen_hash.rs
+    pub fn hash(
```

Comment:
This is great to have the reference, but it's not a good doc for the function. Here it should explain what does it do.